host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -204,6 +205,22 @@ public HostControllerRegistrationHandler(ManagementChannelHandler handler, Domai
          */
         ModelNode executeReadOnly(ModelNode operation, Resource model, OperationStepHandler handler, ModelController.OperationTransactionControl control);
 
+        /**
+         * Registers a slave host controller. A non-exclusive read lock is held for the duration of this registration
+         * process, blocking writes, but allowing the initiation of additional read locks.
+         * @param operationId - the operation ID for this registration
+         * @param hostInfo - the registering host's HostInfo
+         * @throws OperationFailedException - if the lock acquisition fails
```

Comment:
OperationFailedException is an "Exception indicating an operation has failed due to a client mistake (e.g. an operation with invalid parameters was invoked.) Should not be used to report server failures." The failures here indicate some sort of server problem (e.g. an operation had no id, or a thread was interrupted).

I don't want to use OFE for any other purpose because lots of devs, particularly subsystem authors, tend to misuse it. So I don't want examples of that in the kernel code.

Perhaps these can just throw IAE (for invalid/null params) and InterruptedException.